php_pthreads.c

```diff
@@ -156,11 +156,11 @@ static inline void pthreads_execute_ex(zend_execute_data *data) {
 /* {{{ */
 static inline zend_bool pthreads_verify_type(zend_execute_data *execute_data, zval *var, zend_arg_info *info) {
 	pthreads_object_t *threaded = NULL;
-
+        
 	if (!var ||
 #if PHP_VERSION_ID >= 70200
-		!ZEND_SAME_FAKE_TYPE(ZEND_TYPE_CODE(info->type), Z_TYPE_P(var)) &&
-		!ZEND_TYPE_IS_CLASS(info->type) ||
+                Z_ISNULL_P(var) ||
```

Comment:
Would it not be enough to simply change the `&&` into an `||`? The current logic of:
```
!ZEND_SAME_FAKE_TYPE(ZEND_TYPE_CODE(info->type), Z_TYPE_P(var)) &&
!ZEND_TYPE_IS_CLASS(info->type)
```

doesn't work, because if it is not the same (fake) type, then the anything-but-a-class check will always fail. If the type is not a class, then we want to (and must) short circuit, since a class type is presumed when the `instanceof_function` function is used for the next check. Or am I missing something?